python/common/org/python/types/Set.java

```diff
@@ -416,9 +455,12 @@ public boolean __setattr_null(java.lang.String name, org.python.Object value) {
             args = {"other"}
     )
     public org.python.Object intersection(org.python.Object other) {
-        java.util.Set set = ((Set) this.copy()).value;
-        set.retainAll(((Set) other).value);
-        return new Set(set);
+        try {
+            org.python.types.Set otherSet = new org.python.types.Set(new org.python.Object[] {other}, null);
```

Comment:
It would be a lot more efficient to check whether `other` is already a set - and if it is, pass it unchanged to `__and__`, rather than universally duplicating it.